kubevirtci

```diff
@@ -185,6 +188,10 @@ function kubevirtci::create_cluster() {
 
 	echo "Waiting for worker VM in tenant cluster namespace"
 	kubevirtci::retry_until_success kubevirtci::vm_matches "${TENANT_CLUSTER_NAME}-md-"
+
+	if [ $tmp_template ]; then
+		rm -rf $tmp_template
+	fi
```

Comment:
this if statement isn't checking if the variable is set or not. i don't think it will ever execute

Reply:
So AFAIK this does check if $tmp_template has a value, if so, the if statement is executed, you can check this easily with this bash script
```bash
#!/usr/bin/env bash

if [ $test ]; then
  echo "test defined as $test"
else
  echo "not defined"
fi
```
This should return not defined, then before the if set it to a value, it will then echo `test defined as value`

Reply:
I can use `-v` if that makes it clearer, the result will be the same.

Reply:
```
#!/bin/bash
tmpdir=$(mkdir tmpxxxx)
echo "created $tmpdir"
if [ $tmpdir ]; then
        echo "removing $tmpdir"
        rm -rf $tmpdir
fi
```

the echo within the if statement isn't executed

Reply:
it works because the variable is not empty, that is why the mkdir example didn't work, because that didn't return the name of the directory it created unlike mktemp.